### ISSUE TITLE
[FEAT] Add simple cluster metrics

### ISSUE THREAD

**Issue by RossKen** [moj-analytical-services/splink#1538]
### Is your proposal related to a problem?

We need better ways of looking at the distributions of our cluster, which can then be used to select which metrics go into cluster studio. Simple metrics, such as cluster size and cluster density, should provide a good starting point.

### Describe the solution you'd like

Add simple graph metrics such as cluster size and cluster density to Splink outside of cluster studio. Probably worth creating some charts as well to go alongside.

**Comment by samnlindsay** [comment]
Example Spark SQL query to take edges and clusters and add size and density:
```sql
WITH cluster_stats AS (
    SELECT
        count(*) as n_edges,
        size(array_union(collect_set(e.src), collect_set(e.dst))) as n_nodes,
        cc.component as cluster_medium
    FROM edges e
    LEFT JOIN cc_medium as cc
    ON e.src = cc.id
    GROUP BY cc.component
)

SELECT
    cluster_medium,
    n_nodes,
    n_edges,
    n_edges * 0.5 / (n_nodes * (n_nodes-1)) as density
FROM cluster_stats
WHERE n_nodes > 4
ORDER BY density
```

**Comment by ADBond** [comment]
Related: #1001, also possibly #539

**Comment by ADBond** [comment]
Closed by #2027.

Further related work still to be done - see #1001
